content/updates/terraform-null-label.md

```diff
@@ -0,0 +1,194 @@
+
+The issue of inconsistent naming becomes even more pronounced when we consider tags. What was once simply "backend" now has new variations like "backend-auth" and "backend-job-processor". Are those cloud resources getting tagged correctly so whoever is paying the bill knows that the "job processor" is costing 10x what the "auth" service is? Probably not.
+
+As the stack expands and you need to hire more engineers, the cognitive load and spool-up time for each new hire grows and grows. As it becomes harder to understand and categorize and "know about" the infrastructure, it becomes a bigger security/attack surface. Eventually billing and resource attribution becomes a problem too.
+
+## The Fix: terraform-null-label
+
+Well, the **real** fix is consistent naming and tagging, but the challenge has always been getting a growing, changing roster of engineers to decide on, implement, and enforce a standard _over time_.
+
+Good luck with that. Maybe you’ve written down your standards in documentation, but that’s hard to find, rarely read, and not easily enforceable.
+
+Why not use code?
+
+The `terraform-null-label` module, available at [https://github.com/cloudposse/terraform-null-label](https://github.com/cloudposse/terraform-null-label), offers a much better solution. We’ll look at some example usage in the next section but at a high level It allows users to define up to six primary inputs, referred to as "ID elements". These include:
+
+1. **Namespace:** A short abbreviation of the company name, typically 2-4 letters.
+2. **Tenant:** This is used for resources dedicated to a specific customer or team. (For a team like "data analytics" => "da")
+3. **Environment:** This refers to the AWS region. (Note that this nomenclature may be inverse to what most engineers are used to).
+4. **Stage:** This indicates the stage of the application, such as `dev`, `stage`, `prod`. This is what we would normally call an `environment` in most settings, but legacy naming within the module has made this a bit funky.
+5. **Name:** This is the name of the component or service, such as 'rds','eks', or `backend-auth`.
+6. **Attributes:** This is an arbitrary list of strings that can be used to add additional descriptors to the final ID.
+
+The `terraform-null-label` module provides a standardized set of outputs that combine these various ID elements for easy usage across resources and modules. For example, if the following ID elements were supplied to a label module named `frontend_label`:
+
+```hcl
+  namespace  = "mp" # Shorthand for Masterpoint
+  environment = "uw2" # Shorthand for the AWS us-west-2 region
+  stage      = "prod"
+  name       = "frontend"
+  attributes = ["public"]
```

Comment:
```suggestion
```suggestion
  namespace   = "mp" # Shorthand for Masterpoint
  environment = "uw2" # Shorthand for the AWS us-west-2 region
  stage       = "prod"
  name        = "frontend"
  attributes  = ["public"]
```